lib/api/bgppeer.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	. "github.com/tigera/libcalico-go/lib/api/unversioned"
+	. "github.com/tigera/libcalico-go/lib/net"
+)
+
+
+type BGPPeerMetadata struct {
+	ObjectMetadata
+	// The hostname of the node that is peering with this peer.  If left blank,
+	// Calico assumes this is a "global" BGP peer - i.e. it peers with every Calico
+	// node in the deployment.
+	Hostname string `json:"hostname,omitempty" validate:"omitempty,name"`
+	// The IP address of the peer.
+	PeerIP   IP     `json:"peerIP" validate:"omitempty,ip"`
```

Comment:
Oooh - I'm glad you spotted that!   My original thought process was we need to be able to handle the Spec fields not being assigned for a get or a delete!  But that just doesn't make sense since we don't validate the Spec on a get or a delete.  I will remove!  That covers the ASNumber field.

Now the IP address field is interesting.  That is part of the Metadata - and arguably we should validate the Metadata - but a get needs to allow fields being missing from the Metadata - so it'll list rather than get a single value.  But in hindsight again - I think we just don't want to validate this on a get at all - therefore I'll remove for the IP field as well.  I'll check the other resource types for this too.